zyte_common_items/processors.py

```diff
@@ -104,50 +107,78 @@ def _from_zp_breadcrumb(value: zp_Breadcrumb) -> Breadcrumb:
     return results
 
 
-@only_handle_nodes
-def brand_processor(value: Union[Selector, HtmlElement], page: Any) -> Any:
+def brand_processor(value: Any, page: Any) -> Any:
     """Convert the data into a brand name if possible.
 
-    Supported inputs are :class:`~parsel.selector.Selector`,
-    :class:`~parsel.selector.SelectorList` and :class:`~lxml.html.HtmlElement`.
-    Other inputs are returned as is.
+    If inputs are either :class:`~parsel.selector.Selector`,
+    :class:`~parsel.selector.SelectorList` or :class:`~lxml.html.HtmlElement`, attempts
+    to extract brand data from it.
+
+    If value is a string, uses it to create a :class:`~zyte_common_items.Brand` instance.
+
+    Other inputs are returned unchanged.
     """
-    return extract_brand_name(value, search_depth=2)
+    value = _handle_selectorlist(value)
 
+    if isinstance(value, str):
+        return Brand(name=value) if value else None
```

Comment:
@kmike Do you think stripping would make sense here? Or if we want to keep the original spacing for non-empty strings, at least do `… if value.strip() else …`?

Reply:
I think stripping is fine here.

Reply:
Added `strip()` and some more test cases in 12cb8efadd59d399b41bf59de6b0d721d4d10efb